### DISCUSSION TITLE
Feedback on new user interface and tutorials (pyro & scvi-tools)

### DISCUSSION THREAD

**Discussion by vitkl** [BayraktarLab/cell2location#65] [Ideas]
We are finalising a major revision of cell2location including backend (scvi-tools & pyro) and user interface. Would be good to get any thoughts or feedback on the new interface and the new tutorial.

**Comment by edward130603** [comment]
I've been using the new version of cell2location recently and it's been much easier to adapt the tutorial to my data this time compared to before. A few suggestions that I think would improve clarity for users:

1. In block 5, my gene ids were not named 'Gene-ID2'. A comment here to replace with your gene name column in user's own data would be helpful.
2. In block 6, it would be nice to have some insight on how to select the parameters for gene filtering.
3. In block 8, similarly, would be nice to understand the tuning parameters. I needed to increase epochs to ~400 to get stabilization of loss, but not sure about what other params do here.
4. For block 10, any additional guidance on what to expect from reconstruction accuracy plot?
5. Raw code before blocks 11/15, I think `mod` should be defined after `adata` is defined.
6. For block 40, would it be possible to put `plot_genes_per_cell_type` with the other plotting functions. I think my install did not include `tutorial_utils` but I found it on your github and just copied over the function definition.

**Reply by vitkl** [reply]
Thank you for the comments @edward130603 ! I will change this when updating the tutorial for the next release!

6. I will add a note that users need to copy that code. That code will often need customization, so I don't think it makes sense to add it to the package.
